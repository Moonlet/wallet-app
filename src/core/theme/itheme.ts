export interface ITheme {
    dark: boolean;
    shadowGradient: string[];

    fontSize: {
        small: number;
        regular: number;
        large: number;
    };

    colors: {
        primary: string;
        accent: string;
        text: string;
        textSecondary: string;
        textTertiary: string;
        positive: string;
        negative: string;
        error: string;
        warning: string;

        settingsDivider: string;
        cardBackground: string;
        appBackground: string;
        inputBackground: string;
        disabledButton: string;
        modalBackground: string;

        gradientLight: string;
        gradientDark: string;
    };
}
