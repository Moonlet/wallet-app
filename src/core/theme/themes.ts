export enum Theme {
    dark = 'dark',
    light = 'light'
}
