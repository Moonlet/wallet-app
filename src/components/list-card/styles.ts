import { StyleSheet } from 'react-native';
import { BASE_DIMENSION, BORDER_RADIUS, ICON_CONTAINER_SIZE } from '../../styles/dimensions';
import { ITheme } from '../../core/theme/itheme';

export default (theme: ITheme) =>
    StyleSheet.create({
        card: {
            display: 'flex',
            flexDirection: 'row',
            alignItems: 'center',
            padding: BASE_DIMENSION * 2,
            backgroundColor: theme.colors.cardBackground,
            borderRadius: BORDER_RADIUS,
            marginBottom: BASE_DIMENSION,
            borderWidth: 1,
            borderColor: theme.colors.cardBackground
        },
        labelContainer: {
            flex: 1
        },
        label: {
            flex: 1,
            fontSize: 18,
            lineHeight: 25,
            fontWeight: '500',
            letterSpacing: 0.38,
            color: theme.colors.text,
            opacity: 0.87
        },
        selected: {
            borderColor: theme.colors.accent
        },
        iconLeftContainer: {
            height: ICON_CONTAINER_SIZE,
            width: ICON_CONTAINER_SIZE,
            alignItems: 'flex-start',
            justifyContent: 'center'
        },
        iconRightContainer: {
            height: ICON_CONTAINER_SIZE,
            width: ICON_CONTAINER_SIZE,
            alignItems: 'flex-end',
            justifyContent: 'center'
        },
        icon: {
            color: theme.colors.accent
        }
    });
