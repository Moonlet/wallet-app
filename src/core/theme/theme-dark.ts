import { ITheme } from './itheme';

// this should be moved in styles folder

export const darkTheme: ITheme = {
    dark: true
};
