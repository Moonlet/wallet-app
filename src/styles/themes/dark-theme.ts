import { ITheme } from '../../core/theme/itheme';
import { COLORS } from '../colors';

export const darkTheme: ITheme = {
    dark: true,
    shadowGradient: ['#00000000', '#000000CC'],

    fontSize: {
        small: 12,
        regular: 16,
        large: 34
    },

    colors: {
        primary: COLORS.GRAY,
        accent: COLORS.AQUA,
        text: COLORS.WHITE,
        textDarker: COLORS.LIGHT_GRAY,
        positive: COLORS.HOT_GREEN,
        negative: COLORS.HOT_RED,

        cardBackground: COLORS.DARK_GRAY,
        appBackground: COLORS.COD_GRAY,
        settingsDivider: COLORS.BLACK
    }
};
