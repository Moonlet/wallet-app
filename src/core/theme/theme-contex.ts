import React from 'react';
import { ITheme } from './itheme';
import { darkTheme } from '../../styles/themes/dark-theme';

export const ThemeContext = React.createContext<ITheme>(darkTheme);
