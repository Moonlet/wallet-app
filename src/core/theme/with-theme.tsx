import React from 'react';
import { ThemeContext } from './theme-contex';
import { ITheme } from './itheme';

export const withTheme = (styleProvider: (theme: ITheme) => any): any => (Comp: any) => {
    function Component(props: any) {
        const theme = React.useContext(ThemeContext);
        return (
            <Comp {...props} styles={styleProvider(theme)} theme={theme}>
                {props.children}
            </Comp>
        );
    }
    Component.displayName = `withTheme(${Comp.displayName || Comp.name || 'Component'})`;
    return Component;
};
