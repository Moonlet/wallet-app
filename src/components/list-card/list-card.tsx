import React from 'react';
import { Text } from '../../library';
import { smartConnect } from '../../core/utils/smart-connect';
import { withTheme, IThemeProps } from '../../core/theme/with-theme';
import stylesProvider from './styles';
import { View, TouchableHighlight } from 'react-native';
import Icon from '../icon/icon';
import { ICON_SIZE, normalize, BASE_DIMENSION } from '../../styles/dimensions';
import { IconValues } from '../icon/values';
import { SmartImage } from '../../library/image/smart-image';

export interface IProps {
    label: string | JSX.Element;
    leftIcon?: IconValues | React.ComponentType<any>;
    rightIcon?: string;
    selected?: boolean;
    onPress?: any;
    style?: any;
}

export const ListCardComponent = (
    props: IProps & IThemeProps<ReturnType<typeof stylesProvider>>
) => {
    const label =
        typeof props.label === 'string' ? (
            <Text style={props.styles.label}>{props.label}</Text>
        ) : (
            props.label
        );

    return (
        <TouchableHighlight
            testID={String(props.label)
                .replace(/ /g, '-')
                .toLowerCase()}
            onPress={props.onPress}
            underlayColor={props.theme.colors.appBackground}
        >
            <View style={[props.styles.card, props.selected && props.styles.selected, props.style]}>
                {props.leftIcon ? (
                    props.leftIcon.toString().includes('data:image/') ? (
                        <SmartImage
                            source={{ iconComponent: props.leftIcon as React.ComponentType }}
                            style={{ marginRight: BASE_DIMENSION * 2 }}
                        />
                    ) : (
                        <View style={[props.styles.iconContainer, { alignItems: 'flex-start' }]}>
                            <Icon
                                name={props.leftIcon as string}
                                size={ICON_SIZE}
                                style={props.styles.icon}
                            />
                        </View>
                    )
                ) : null}

                <View style={props.styles.labelContainer}>{label}</View>

                {props.rightIcon && (
                    <View style={[props.styles.iconContainer, { alignItems: 'flex-end' }]}>
                        <Icon
                            name={props.rightIcon}
                            size={normalize(18)}
                            style={props.styles.icon}
                        />
                    </View>
                )}
            </View>
        </TouchableHighlight>
    );
};

export const ListCard = smartConnect<IProps>(ListCardComponent, [withTheme(stylesProvider)]);
