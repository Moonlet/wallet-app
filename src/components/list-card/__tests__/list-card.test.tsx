import 'react-native';
import React from 'react';
import { ListCardComponent, IProps } from '../list-card';
import stylesProvider from '../styles';

import { shallow } from 'enzyme';
import { darkTheme } from '../../../styles/themes/dark-theme';
import { IThemeProps } from '../../../core/theme/with-theme';

const props: IProps & IThemeProps<ReturnType<typeof stylesProvider>> = {
    label: 'card',
    leftIcon: 'money-wallet-1',
    rightIcon: 'chevron-right',
    selected: true,
    onPress: jest.fn(),
    styles: stylesProvider(darkTheme),
    theme: darkTheme
};

describe('CardList component', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<ListCardComponent {...props} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
