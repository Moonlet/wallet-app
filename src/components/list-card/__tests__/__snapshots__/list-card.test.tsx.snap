// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList component renders correctly 1`] = `
"<TouchableHighlight testID=\\"card\\" onPress={[Function: mockConstructor]} underlayColor=\\"#121212\\" activeOpacity={0.85} delayPressOut={100}>
  <View style={{...}}>
    <View style={{...}}>
      <Icon name=\\"money-wallet-1\\" size={44} style={{...}} allowFontScaling={false} />
    </View>
    <View style={{...}}>
      <withTheme(TextComponent) style={{...}}>
        card
      </withTheme(TextComponent)>
    </View>
    <View style={{...}}>
      <Icon name=\\"chevron-right\\" size={33} style={{...}} allowFontScaling={false} />
    </View>
  </View>
</TouchableHighlight>"
`;
